ensemble-view/src/main/java/hec/ensembleview/PlotStatisticsForChartType.java

```diff
@@ -0,0 +1,206 @@
+        db = database;
+        zdt = zonedDateTime;
+        rid = recordIdentifier;
+    }
+
+    public void addStatisticsToProbabilityPlot(EnsembleChartAcrossEnsembles chart, EnsembleViewStat[] stats, EnsembleTimeSeries ets) {
+        for (EnsembleViewStat selectedStat : stats) {
+            Statistics statType = selectedStat.getStatType();
+            switch (statType) {
+                case MIN:
+                case MAX:
+                case AVERAGE:
+                case MEDIAN:
+                case MAXACCUMDURATION:
+                case MAXAVERAGEDURATION:
+                case TOTAL:
+                    chart.addProbPoint(
+                            new PointSpec(0,
+                                    StatComputationHelper.computeProbability(ets, statType, zdt, ChartType.SCATTERPLOT),
+                                    getStrokeForStatType(statType),
+                                    getColorForStatType(statType),
+                                    StatisticsStringMap.map.get(selectedStat.getStatType())));
+                    break;
+                case PERCENTILE:
+                    float[] percentiles = ((TextBoxStat) selectedStat).getTextFieldValue();
+                    DecimalFormat df = new DecimalFormat("0.0");
+
+                    for(int i = 0; i < percentiles.length; i++) {
+                        chart.addProbPoint(
+                                new PointSpec(0, StatComputationHelper.computeProbability(ets, statType, zdt, new float[] {(percentiles[i])}, ChartType.SCATTERPLOT),
+                                        getStrokeForStatType(statType), randomColor(i+100), StatisticsStringMap.map.get(selectedStat.getStatType()) + " " + df.format(percentiles[i]*100) + "%"));
+                    }
+                    break;
+                case NDAYCUMULATIVE:
+                    float[] accumulatedDays = ((TextBoxStat) selectedStat).getTextFieldValue();
+
+                    for(int i = 0; i< accumulatedDays.length; i++) {
+                        chart.addProbPoint(
+                                new PointSpec(1, StatComputationHelper.computeProbability(db.getEnsembleTimeSeries(rid), statType, zdt, new float[] {(accumulatedDays[i])}, ChartType.SCATTERPLOT),
+                                        getStrokeForStatType(statType), randomColor(i + 100), StatisticsStringMap.map.get(selectedStat.getStatType()) + accumulatedDays[i] + " days"));
+                    }
+                    break;
+            }
+        }
+    }
+
+    public void addStatisticsToTimePlot(EnsembleChartAcrossTime chart, EnsembleViewStat[] stats, EnsembleTimeSeries ets, ZonedDateTime[] dates) throws Exception {
+        for (EnsembleViewStat selectedStat : stats) {
+
+            Statistics statType = selectedStat.getStatType();
+            float[] percentiles = null; // Initialize percentiles if needed
```

Comment:
It looks like this variable is only referenced within the PERCENTILE case. Can it be declared there instead?